src/utils/notifications.ts

```diff
@@ -0,0 +1,144 @@
+import MagicBellClient, { Notification } from '@magicbell/core';
+import { query } from '~amplifyClient';
+import {
+  GetColonyContributorsNotificationDataDocument,
+  GetColonyContributorsNotificationDataQuery,
+  GetColonyContributorsNotificationDataQueryVariables,
+  NotificationsDataFragment,
+} from '~graphql';
+
+export enum NotificationType {
+  SimplePayment = 'SimplePayment',
+  Mention = 'Mention',
+}
+
+interface NotificationVariables {
+  title: string;
+  creator: string;
+  colonyAddress: string;
+  transactionHash: string;
+  notificationType: NotificationType;
+  amount?: string;
+  tokenAddress?: string;
+  mentions?: string[];
+}
+
+// Set up the notification client
+export const setupNotificationsClient = async (): Promise<void> => {
+  await MagicBellClient.createInstance({
+    apiKey: process.env.MAGICBELL_API_KEY as string,
+    apiSecret: process.env.MAGICBELL_API_SECRET as string,
+  });
+};
+
+// Check if notifications should be sent to a recipient from a colony.
+const shouldSendNotificationToRecipient = (
+  notificationsData: NotificationsDataFragment,
+  colonyAddress: string,
+): boolean => {
+  return (
+    !!notificationsData.magicbellUserId && // User has a magicbell user account made
+    !notificationsData.notificationsDisabled && // User has not disabled notifications app-wide
+    !notificationsData.mutedColonyAddresses.includes(colonyAddress) // User has not muted the colony that the notification is in
+  );
+};
+
+// Get all the recipients of a colony wide notification in the format Magicbell expects.
+const getRecipientsOfColonyWideNotification = async (
+  colonyAddress: string,
+): Promise<Array<{ external_id: string }>> => {
+  const recipients: Array<{ external_id: string }> = [];
+  let nextToken: string | null | undefined;
+
+  do {
+    const members = await query<
+      GetColonyContributorsNotificationDataQuery,
+      GetColonyContributorsNotificationDataQueryVariables
+    >(GetColonyContributorsNotificationDataDocument, {
+      colonyAddress,
+      limit: 100,
+      ...(nextToken ? { nextToken } : {}),
+    });
+
+    if (members?.data?.getContributorsByColony) {
+      members.data.getContributorsByColony.items.forEach((member) => {
+        const notificationsData = member?.user?.notificationsData;
+
+        if (!notificationsData) {
+          return;
+        }
+
+        if (
+          shouldSendNotificationToRecipient(notificationsData, colonyAddress)
+        ) {
+          recipients.push({
+            external_id: notificationsData.magicbellUserId,
+          });
+        }
+      });
+
+      nextToken = members?.data?.getContributorsByColony?.nextToken;
+    } else {
+      nextToken = null;
+    }
+  } while (nextToken);
+
+  return recipients;
+};
+
+// Send a notification when an action is made.
+export const sendActionNotification = async ({
```

Comment:
should we maybe split this into a 
`sendColonyWideNotification` which will send it to all recipients (we can filter out the mentions array if that type of notifications exists)
`sendActionNotification` which will send notifications directly for the impacted user addresses? Then we could also query notification data just for those users instead of always getting notification data for all colony contributors?

Reply:
So actually on every single action we will be sending a colony wide notification, but the mention will be optional, which is why I structured it this way. I have created a new `createNotification` function which can now be used to just send a straight up notification if we need to do that at all though, skipping all the contributor fetching steps.